spring-cloud-gcp-bigquery/src/main/java/com/google/cloud/spring/bigquery/core/BigQueryTemplate.java

```diff
@@ -159,16 +181,152 @@ public ListenableFuture<Job> writeDataToTable(
   }
 
   /**
-   * @return the name of the BigQuery dataset that the template is operating in.
+   * This method uses BigQuery Storage Write API to write new line delimited JSON file to the
+   * specified table. This method creates a table with the specified schema.
+   *
+   * @param tableName name of the table to write to
+   * @param jsonInputStream input stream of the json file to be written
+   * @return {@link ListenableFuture} containing the WriteApiResponse indicating completion of
+   *     operation
    */
+  @Override
+  public ListenableFuture<WriteApiResponse> writeJsonStream(
+      String tableName, InputStream jsonInputStream, Schema schema) {
+    createTable(tableName, schema); // create table if it's not already created
```

Comment:
This sounds like a relatively lengthy operation -- are there timeout configurations we can expose to the users later?

Reply:
Definitely, I have added it as a feedback @ go/bq-write-api-sc-gcp-integration. We will look into it later.